vpn-ca/main.go

```diff
@@ -147,12 +146,16 @@ func getTemplate(commonName string, notAfter time.Time, keyUsage x509.KeyUsage,
 	}
 }
 
-func sign(caInfo *caInfo, commonName string, tpl *x509.Certificate) *x509.Certificate {
-	key := generateKey(filepath.Join(caInfo.caDir, fmt.Sprintf("%s.key", commonName)))
+func sign(caInfo *caInfo, commonName string, tpl *x509.Certificate, targetDir string) *x509.Certificate {
+	if _, err := os.Stat(targetDir); os.IsNotExist(err) {
+		os.Mkdir(targetDir, 0644)
```

Comment:
Ah yes, this is the directory permissions. We should probably use `0700` here indeed!